src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -264,6 +264,49 @@ def recover_keyvault(cmd, client, vault_name, resource_group_name, location):
                                    parameters=params)
 
 
+def _parse_network_acls(cmd, resource_group_name, network_acls):
+    if not network_acls:
+        return None
+
+    logger.info('network_acls: {}'.format(network_acls))
+
+    try:
+        network_acls_json = json.loads(network_acls)
+    except:  # pylint: disable=bare-except
+        with open(network_acls) as f:
+            logger.info('network_acls_filename: {}'.format(network_acls))
+            network_acls_json = json.load(f)
+
+    try:
+        network_acls_json = json.loads(json.dumps(network_acls_json))
+    except:  # pylint: disable=bare-except
+        raise CLIError('Please specify a valid JSON to parameter --network-acls')
+
+    logger.info('network_acls_json: {}'.format(network_acls_json))
+
+    VirtualNetworkRule = cmd.get_models('VirtualNetworkRule', resource_type=ResourceType.MGMT_KEYVAULT)
+    IPRule = cmd.get_models('IPRule', resource_type=ResourceType.MGMT_KEYVAULT)
+
+    network_acls = _create_network_rule_set(cmd)
+
+    network_acls.virtual_network_rules = []
+    for vnet_rule in network_acls_json.get('vnet', []):
+        items = vnet_rule.split('/')
+        if len(items) != 2:
+            raise CLIError('Invalid VNet rule: {}. Format: {{vnet}}/{{subnet}}'.format(vnet_rule))
```

Comment:
can user input resource id directly?

Reply:
> can user input resource id directly?

@yungezz It should be supported, I will update the code.